cmake/AwsFindPackage.cmake

```diff
@@ -0,0 +1,22 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+option(IN_SOURCE_BUILD "If the CRT libs are being built from your source tree (add_subdirectory), set this to ON" OFF)
+
+# This function handles dependency list building based on if traditional CMAKE modules via. find_package should be
+# used, vs if this is an in source build via. something like git submodules and add_subdirectory.
+# This is largely because CMake was not well planned out, and as a result, in-source and modules don't play well
+# together. Only use this on CRT libraries (including S2N), libcrypto will stay as an assumed external dependency.
```

Comment:
This is rather tortured English.  And, presumably there is no need to comment on whether CMake, etc. was "well planned" or not.  And, it's unclear (at least to me) what you mean by "CRT libraries" (i.e., presumably you don't mean the C Language Run-Time Library or any of the analogous C++ Language ones).

How about something like:
> This function handles building a list of external dependencies, taking into account whether the dependency is built as a submodule or found elsewhere on the build platform.  It should be used as a replacement for `find_package()` in cases where the dependency might satisfied either by using an installed library or by building it as part of the build.  The `IN_SOURCE_BUILD` option determines how the dependency will be satisfied.  For dependencies which are never satisfied as part of the build (such as `libcrypto`), `find_package()` should be used instead of this function.

Reply:
This file is from https://github.com/awslabs/aws-c-common/blob/master/cmake/AwsFindPackage.cmake
If you want me to update the comment I can add to this pr.

Reply:
Sorry -- Github's presentation made it look like this was a new file that you were adding, so I figured it would be best if it had better verbiage before it was merged in.

In any case, I was just making a suggestion -- I'm not an `aws-sdk-cpp` maintainer, so I'll defer to them on what they want.